File: Bot/Commands/CommandHandler.py
from telethon import TelegramClient
from telethon.tl.types import Message

from apihelper.BotSettingsServiceApi import BotSettingsServiceApi
from Models.BotSettingsModel import BotSettingsModel
from utility.inline_replies import *


class CommandHandler:
    def __init__(self, bot_client: TelegramClient , bot_settings_service : BotSettingsServiceApi):
        self.bot_client = bot_client
        self.bot_settings_service = bot_settings_service
        
        self.bot_settings = self.bot_settings_service.botsettings

    async def on_set_main_group(self, message: Message):
        chat_id = message.chat_id

        self.bot_settings = self.bot_settings_service.get_bot_settings()
        
        self.bot_settings["mainGroupID"] = chat_id
        self.bot_settings["mainGroupUsername"] = message.chat.title

        self.bot_settings = self.bot_settings_service.set_bot_settings(self.bot_settings)

        # log_message = f"Set Main Group: {message.chat.title}, {chat_id}"
        # self.logger.log_info(log_message)

        await self.bot_client.send_message(
            chat_id,
            message="این گروه به عنوان گروه اصلی ثبت شد"
        )

    async def on_set_main_channel(self, message: Message):
        message_text = message.text
        chat_id = message.chat_id

        channel_id = message_text.split("\n")[1]
        main_channel = await self.bot_client.get_entity(channel_id)

        self.bot_settings = self.bot_settings_service.get_bot_settings()
        
        self.bot_settings["mainChanneID"] = int(f"-100{main_channel.id}")
        self.bot_settings["mainChannelUsername"] = main_channel.username

        self.bot_settings = self.bot_settings_service.set_bot_settings(self.bot_settings)

        await self.bot_client.send_message(
            chat_id,
            message=f"کانال {channel_id} به عنوان کانال اصلی ثبت شد"
        )

    async def on_set_inspect_topic(self, message: Message):
        topic_id = message.reply_to_msg_id
        if topic_id:
            chat_id = message.chat_id

            inline_keyboard = set_inspect_topic_inline_keyboard(chat_id, topic_id)

            await self.bot_client.send_message(
                chat_id,
                message="برای انتخاب این تاپیک به عنوان تاپیک بازرسی تایید کنید",
                buttons=inline_keyboard,
                reply_to=topic_id
            )

    async def on_set_license_check_topic(self, message: Message):
        topic_id = message.reply_to_msg_id
        if topic_id:
            chat_id = message.chat_id

            inline_keyboard = set_license_check_topic_inline_keyboard(chat_id, topic_id)

            await self.bot_client.send_message(
                chat_id,
                message="برای انتخاب این تاپیک به عنوان تاپیک بررسی مجوز تایید کنید",
                buttons=inline_keyboard,
                reply_to=topic_id
            )

    async def on_set_main_alternative_group_topic(self, message: Message):
        topic_id = message.reply_to_msg_id
        if topic_id:
            chat_id = message.chat_id

            inline_keyboard = set_main_alternative_group_topic_inline_keyboard(chat_id, topic_id)

            await self.bot_client.send_message(
                chat_id,
                message="برای انتخاب این تاپیک به عنوان تاپیک ارسال پست ها تایید کنید",
                buttons=inline_keyboard,
                reply_to=topic_id
            )

File: BackEnd/TelegramBot.Bot/CommandHandler.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using AutoMapper;
using Telegram.Bot;
using Telegram.Bot.Types;
using Telegram.Bot.Types.ReplyMarkups;
using TelegramBot.Application.Interfaces;
using TelegramBot.Application.Services.BotSettings;
using TelegramBot.Application.Services.PostService;
using TelegramBot.Core.Entities;

namespace TelegramBot.Bot
{
    public class CommandHandler
    {
        IMapper mapper;
        BotSettingsEntity botSettings;
        IBotSettingsService botSettingsService;
        ILoggerService logger;
        private readonly ITelegramBotClient botClient;
        private readonly Update update;
        private readonly CancellationToken cancellationToken;

        public CommandHandler(ITelegramBotClient botClient, Update update, CancellationToken cancellationToken, BotSettingsEntity botSettings, ILoggerService logger, IMapper mapper, IBotSettingsService botSettingsService)
        {
            this.botClient = botClient;
            this.update = update;
            this.cancellationToken = cancellationToken;
            this.botSettings = botSettings;
            this.logger = logger;
            this.mapper = mapper;
            this.botSettingsService = botSettingsService;
        }


        public async void OnSetMainGroup(Message message)
        {
            var chatId = message.Chat.Id;

            var newbotSettings = mapper.Map<BotSettingsModel>(botSettings);

            newbotSettings.MainGroupID = chatId;
            newbotSettings.MainGroupUsername = update.Message.Chat.Title;

            botSettingsService.SetBotSettings(newbotSettings);

            var logMessage = $"Set Main Group: {update.Message.Chat.Title} , {chatId}";
            logger.LogInfo(logMessage);

            Message setMainChannel = await botClient.SendTextMessageAsync(
                chatId: chatId,
                text: "این گروه به عنوان گروه اصلی ثبت شد",
                cancellationToken: cancellationToken);
        }

        public async void OnSetMainChannel(Message message)
        {
            var messageText = message.Text;
            var chatId = message.Chat.Id;

            var channelId = messageText.Split("\n");

            Chat mainChannel = await botClient.GetChatAsync($"{channelId[1]}");

            var newbotSettings = mapper.Map<BotSettingsModel>(botSettings);
            newbotSettings.MainChanneID = mainChannel.Id;
            newbotSettings.MainChannelUsername = mainChannel.Username;


            botSettingsService.SetBotSettings(newbotSettings);

            var logMessage = $"Set Main Channel: {mainChannel.Username} , {mainChannel.Id}";
            logger.LogInfo(logMessage);

            Message setMainChannel = await botClient.SendTextMessageAsync(
                chatId: chatId,
                text: $" کانال {channelId[1]} به عنوان کانال اصلی ثبت شد ",
                cancellationToken: cancellationToken);
        }

        public async void OnSetInspectTopic(Message message)
        {
            if (message.MessageThreadId is { } topicId)
            {
                var chatId = message.Chat.Id;

                var inlineKeyboard = Utility.setInspectTopic_InlineKeyboard(chatId, topicId);

                Message markupMessage = await botClient.SendTextMessageAsync(
                                            chatId: chatId,
                                            text: "برای انتخاب این تاپیک به عنوان تاپیک بازرسی تایید کنید",
                                            replyMarkup: inlineKeyboard,
                                            replyToMessageId: topicId,
                                            cancellationToken: cancellationToken);
            }
        }

        public async void OnSetLicenseCheckTopic(Message message)
        {
            if (message.MessageThreadId is { } topicId)
            {
                var chatId = message.Chat.Id;

                var inlineKeyboard = Utility.setLicenseCheckTopic_InlineKeyboard(chatId, topicId);

                Message markupMessage = await botClient.SendTextMessageAsync(
                                            chatId: chatId,
                                            text: "برای انتخاب این تاپیک به عنوان تاپیک بررسی مجوز تایید کنید",
                                            replyMarkup: inlineKeyboard,
                                            replyToMessageId: topicId,
                                            cancellationToken: cancellationToken);
            }
        }

        public async void OnSetMainAlternativeGroupTopic(Message message)
        {
            if (message.MessageThreadId is { } topicId)
            {
                var chatId = message.Chat.Id;

                var inlineKeyboard = Utility.setMainAlternativeGroupTopic_InlineKeyboard(chatId, topicId);

                Message markupMessage = await botClient.SendTextMessageAsync(
                                            chatId: chatId,
                                            text: "برای انتخاب این تاپیک به عنوان تاپیک ارسال پست ها تایید کنید",
                                            replyMarkup: inlineKeyboard,
                                            replyToMessageId: topicId,
                                            cancellationToken: cancellationToken);
            }
        }

        public void OnEditedChannelPost()
        {

        }
    }
}